src/widget/widget.cpp

```diff
@@ -1818,7 +1818,13 @@ void Widget::onTryCreateTrayIcon()
             trayMenu->addAction(actionQuit);
             icon->setContextMenu(trayMenu);
 
-            connect(icon, &SystemTrayIcon::activated, this, &Widget::onIconClick);
+            // don't activate qTox widget on tray icon click in Unity backend (see #3419)
+#ifdef ENABLE_SYSTRAY_UNITY_BACKEND
```

Comment:
Please remove that ifdef. It complicates release management for linux by adding requirement to specifically compile for unity and "everything else".